### ISSUE TITLE
EXG-MVR in parallel

### ISSUE THREAD

**Issue by tandreasr** [MODFLOW-USGS/modflow6#2010]
Hi,
here is another issue regarding parallel execution.
When starting the attached model in serial mode or in parallel mode using just 1 MPI rank, all works fine.
But when using 2 MPI ranks like this (inside the project directory):

  "E:\Modflow6\mf6.6.0.dev0_win64par\bin\mpiexec.exe" -np 2 "E:\Modflow6\mf6.6.0.dev0_win64par\bin\mf6.exe" -P

the following error occurs:

ERROR REPORT:
  1. Could not open ".\EXG_1_2.EXG.MVR.CBC" on unit 1040. Specified file
     status REPLACE. Specified file format UNFORMATTED. Specified file access
     STREAM. Specified file action READWRITE. IOSTAT error number 30. STOP
     EXECUTION in subroutine openfile().

OS is Windows 11 and I'm using today's nightly build.

Regards,
Andreas

[__error_exg_mvr.zip](https://github.com/user-attachments/files/16845211/__error_exg_mvr.zip)

**Comment by tandreasr** [comment]
Just to clarify:
the problem occurs due to using BUDGET and BUDGETCSV inside a Exchange-Mover.
When removing those 2 entries from the mover file, all works as expected.

**Comment by mjr-deltares** [comment]
Hi @tandreasr, thanks for reporting. I am out of office for the major part of this week but will have a look at this when I am back.

**Comment by mjr-deltares** [comment]
FYI I can reproduce the error. The easy fix should be to only write budgets for one of the two instances of the mover (each of the two process ranks has one) so the file is not created twice. However, it looks as if the output data in that case is not consistent with the serial run. I will have to delve a little deeper. Tbc.

**Comment by mjr-deltares** [comment]
Update: there are some unrelated issues at play so this is taking a little longer than expected but we should be close

**Comment by tandreasr** [comment]
Thanks for your information :-)

**Comment by mjr-deltares** [comment]
Hi @tandreasr , could you try with the latest development build, it should fix the issue with opening the budget files. (PS I'd be curious to hear for your particular example if the mover budgets are consistent)

**Comment by tandreasr** [comment]
Hi,
yes but it will take till October 21th to get back to you, because I'm currently on vacation :-)
tbc
Regards
Andreas

**Comment by mjr-deltares** [comment]
OK, sounds great, enjoy! I will close this issue for now, please feel free to open a new one when you run into problems again.

**Comment by tandreasr** [comment]
Hi @mjr-deltares,
back from vacation I tested your corrections and now it works - thank you very much!
As for the mover budgets:
the results of the serial model version (without subdomains) and the parallelized model versions (with 2 or 4 subdomains) differ quite significantly (head difference about 15m in maximum for 1st stress period...).
But because it's just a test model from the web and I have quite a few other topics to solve on my plate,
I won't be able to further dive into that problem in the next future.
Should I have the chance to narrow down the problem, I will contact you again :-)
Anyway if you want to have a look into that specific problem
(e.g. due to being a good test case for its huge amount of movers),
just contact me to provide all model versions you need.
Best regards
Andreas

**Comment by mjr-deltares** [comment]
Thanks for reporting back Andreas! I will leave this issue closed then and focus on some other things as well for now.
